cmd/ncproxy/server.go

```diff
@@ -66,6 +67,10 @@ func (s *server) setup(ctx context.Context) (net.Listener, net.Listener, error)
 	gService := newGRPCService(s.cache, s.ncproxyNetworking)
 	ncproxygrpc.RegisterNetworkConfigProxyServer(s.grpc, gService)
 
+	// support the v1 ncproxy api
+	v1Wrapper := newV1ServiceWrapper(gService)
```

Comment:
also, is it possible to gate the API used with build flags, so you need to build a dedicated binary to support the legacy API?
and have that binary print warning that the API is deprecated?

Reply:
As in a dedicated version of ncproxy? I think we could just add a warning about the API now and add a new command to ncproxy that returns the version so that we can check a given binary running. Then later we can just remove the support when we're ready, up the version, and add a note in the release notes. Does that work?

Reply:
im worried that if we add a warning/mix the two, someone may accidentally start relying on it unknowingly, whereas if we make it more difficult to build with the older API, we can prevent misuse
the version command sounds pretty good though...

Reply:
Is it cool if I add this new command in a separate PR?